### PR TITLE
Fix Basic01

### LINKED ISSUE #1269: False error in BASIC01 (B01_NO_CHILD) when testing norid.no
In Zonemaster-Enging v4.7.1 I get an error B01_NO_CHILD in the BASIC01 test. 
"norid.no" does not exist as a DNS zone. Try to test "no" instead.
(https://zonemaster.se/en/result/278d7623557fcd62)

When I test norid.no with v4.7.0, there is only a warning in the DNSSEC tests.
(https://dnscheck.test.norid.no/results/afa7d146dc8ac90c)

According to us, the domain norid.no is set up correctly and is working. I think there might be some bug in the implementation of the BASIC01 test in the new 4.7.1 version of the engine.

Both https://zonemaster.se and https://dnscheck.test.norid.no are running the same Backend and LDNS versions.

br
Richard Persson

### PR DESCRIPTION
**Pull request by tgreenx**
## Purpose

This PR updates Basic01 to fix a bug (regression) introduced in the latest fix release by PR #1249.

## Context

Fixes #1269 

## Changes
- Use functions [Zonemaster::Engine::DNSName::common()](https://github.com/zonemaster/zonemaster-engine/blob/350a939219cd4b5f5b06e5abbedce712e14bce17/lib/Zonemaster/Engine/DNSName.pm#L118-L138) and [Zonemaster::Engine::DNSName::is_in_bailiwick()](https://github.com/zonemaster/zonemaster-engine/blob/350a939219cd4b5f5b06e5abbedce712e14bce17/lib/Zonemaster/Engine/DNSName.pm#L140-L144) instead of [index()](https://perldoc.perl.org/functions/index) to make domain names comparison.

- Refactoring.

## How to test this PR

Tests should pass.

Also:
```
$ zonemaster-cli --show-testcase --test Basic/basic01 --level INFO norid.no
Seconds Level    Testcase       Message
======= ======== ============== =======
   0.02 INFO     UNSPECIFIED    Using version v4.7.1 of the Zonemaster engine.
   9.65 INFO     BASIC01        The parent zone is "no" as returned from name servers "i.nic.no/194.146.106.6;i.nic.no/2001:67c:1010:1::53;njet.norid.no/156.154.101.12;not.norid.no/156.154.100.12;not.norid.no/2001:502:ad09::12;x.nic.no/128.39.8.40;x.nic.no/2001:700:0:412f::40;y.nic.no/193.75.4.22;y.nic.no/2001:8c0:8200:1::2;z.nic.no/158.38.8.133;z.nic.no/2001:700:0:52d:158:38:8:133".
   9.65 INFO     BASIC01        The zone "norid.no" is found.
```

### CODE REVIEW

**Comment by matsduf** [comment]
> Unit tests are failing (on Ubuntu), while it works fine on my computer (Debian). I can see where the error is supposed to be from the logs but unsure as to why it happens yet.

"syntax error at /home/runner/work/zonemaster-engine/zonemaster-engine/lib/Zonemaster/Engine/Test/Basic.pm line 459, near ") <=""

**Review by matsduf — COMMENTED** [review]
I have installed and tested the new code and it seems to behave correctly. The code of method `basic01()` is quite hard to follow. Could it be give some more documentation to make it easier to follow it?

**Review by pnax — APPROVED** [review]
